### PR TITLE
clarify License  to be more standard BSD-3, some doc updates

### PR DESCRIPTION
**Pull request by newville**
This updates the LICENSE file to be more in keeping with standard BSD-3, and updates some of the docs in anticipation of 0.9.12.

### CODE REVIEW

File: doc/installation.rst [hunk 1]
```diff
@@ -21,19 +21,24 @@ Downloading and Installation
...
-Lmfit works with Python versions 2.7 and 3.4 - 3.7. Support for Python 2.6 and
-3.3 ended with lmfit versions 0.9.4 and 0.9.8, respectively. Scipy version 0.17,
-NumPy version 1.10, asteval version 0.9.12, six version 1.10, and uncertainties
-version 3.0 are the minimum requirements.
+Lmfit requires the following Python packages, with versions given:
+   * `six` version 1.10 or higher.
+   * `NumPy` version 1.10 or higher.
+   * `SciPy` version 0.17 or higher.
+   * `asteval` version 0.9.12 or higher.
+   * `uncertainties` version 3.0 or higher.
```
Comment (reneeotten) [inline doc/installation.rst]: you lost the hyperlinks above for the packages, just add the ```_``` after the package name again.

File: doc/installation.rst [hunk 2]
```diff
@@ -21,19 +21,24 @@ Downloading and Installation
...
-In order to run the test suite, either the `pytest`_ package is required. Some
-functionality of lmfit requires the `emcee`_ package, some functionality will
-make use of the `pandas`_, `Jupyter`_, `matplotlib`_, `dill`_, or
-`numdifftools`_ packages if available. We highly recommend each of these
-packages.
+All of these are readily available on PyPI, and should be installed
+automatically if installing with `pip install lmfit`.
+
+In order to run the test suite, the `pytest`_ package is required.  Some
+functionality requiers the `emcee`, `pandas`_, `Jupyter`_, `matplotlib`_,
```
Comment (reneeotten) [inline doc/installation.rst]: ```requiers``` --> ```requires```

Comment (reneeotten) [inline doc/installation.rst]: add hyperlinking again: ``` `emcee` ``` --> ``` `emcee`_ ```

File: THANKS.txt [hunk 3]
```diff
@@ -32,15 +34,14 @@ contribution in size to the existing code) are from:
...
-  by lmfit for many years, and is now an external dependency.
-
+  E. O. Le Bigot wrote the uncertainties package, a version of which was
+  used by lmfit for many years, and is now an external dependency.
 
-Additional patches, bug fixes, and suggestions have come from Christoph Deil,
-Francois Boulogne, Thomas Caswell, Colin Brosseau, nmearl, Gustavo Pasquevich,
-Clemens Prescher, LiCode, Ben Gamari, Yoav Roam, Alexander Stark, Alexandre
-Beelen, and many others.
+Additional patches, bug fixes, and suggestions have come from Fausten
```
Comment (reneeotten) [inline THANKS.txt]: ```Fausten``` --> ```Faustin``` ?

File: doc/faq.rst [hunk 4]
```diff
@@ -100,3 +100,65 @@ How should I cite LMFIT?
...
+change during the fit.  If the `NaN` values come from the data and not the
+calculated model, that should be the case.
+
+
+Common sources of `NaN`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are seeing erros due to `NaN` values, you will need to figure out
+where they are coming from and eliminate them.  It is sometimes difficult
+to tell what causes `NaN` values.  Keeep n mind that all values should be
```
Comment (reneeotten) [inline doc/faq.rst]: ```Keeep n``` --> ```Keep in```

File: doc/faq.rst [hunk 5]
```diff
@@ -100,3 +100,65 @@ How should I cite LMFIT?
...
+numbers when fitting.  Some of the most likely causes of `NaN`s are:
+
+   * taking `sqrt(x)` or `log(x)` where `x` is negative.
+
+   * doing "x**y" where `x` is negative.  Since `y` is real, there will
+     be a fractional component, and a negative number to a fractional
+     exponent is not a real number.
+
+Note that all values should be assumed to be either scalar values or numpy
+arrays of double precision real numbers when fitting.  If you use these in
```
Comment (reneeotten) [inline doc/faq.rst]: this sentence is duplicated, same as on lines 141-143, is that intentional?

File: doc/faq.rst [hunk 6]
```diff
@@ -100,3 +100,65 @@ How should I cite LMFIT?
...
+
+
+Common sources of `NaN`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are seeing erros due to `NaN` values, you will need to figure out
+where they are coming from and eliminate them.  It is sometimes difficult
+to tell what causes `NaN` values.  Keeep n mind that all values should be
+assumed to be either scalar values or numpy arrays of double precision real
+numbers when fitting.  Some of the most likely causes of `NaN`s are:
```
Comment (reneeotten) [inline doc/faq.rst]: the ``` `NaN`s ``` renders incorrectly in the HTML output

**Comment by newville** [comment]
@reneeotten Thanks, I'll fix all the links and try to figure out how to spell NaN for sphinx.  

For the license, I agree in principle with keeping the scipy license if we use their code.  We certainly did borrow some of their code at one point (especially for differential evolution) but I was under the impression that we no longer include code lifted from them.   But it is probably easier just to leave their license in place than try to prove we don't use any of their code.

**Comment by reneeotten** [comment]
@newville looks good! 
I was thinking a bit more about licences: when I added the ampgo method - the code was based on the work from Andrea Gavana as clearly stated in ```lmfit/_ampgo.py```, which includes his website so that's not an issue. From the original Google code website (https://code.google.com/archive/p/ampgo/), it states that the license is MIT. Since we are adding stuff now anyway, let's add this also to our LICENSE file and perhaps juss add the MIT license description to ```lmfit/_ampgo.py``` as well? Do you want to add this yourself in this PR or should I do it in a later one?

**Comment by newville** [comment]
@reneeotten yeah, excellent points.  For ampgo I see that it *says* it was MIT, but not the actual text.   I added that it used the MIT license (but not give the license text -- seems like too much to presume).   Does that seem OK to you?

**Comment by newville** [comment]
OK, I'll merge this and we can move forward.

**Pull request merged**